agent-lib/src/main/java/org/terracotta/angela/agent/kit/LocalKitManager.java

```diff
@@ -107,6 +115,48 @@ public void setupLocalInstall(License license, String kitInstallationPath, boole
     }
   }
 
+  void unlockConcurrentInstall(Path localInstallerPath) {
+    logger.debug(Thread.currentThread().getId() +" unlock");
+    File file = localInstallerPath.resolve(ANGELA_LOCK_FILE).toFile();
+    final boolean deleted = file.delete();
+    if (!deleted) {
+      logger.error("Angela Installer lock file can not be deleted when unlocking at {}", file.getAbsolutePath());
+    }
+  }
+
+  void lockConcurrentInstall(Path localInstallerPath) {
+    logger.debug(Thread.currentThread().getId() +" lock");
+    File file = localInstallerPath.resolve("angela-install.lock").toFile();
+    try {
+      if (!file.createNewFile()) {
```

Comment:
the process that gets the lock will release it eventually regardless of the success or not of the kit install so you can not count that the first one to get the lock will have installed successfully the installer 

about using just a synchronized statement, it would work only if all executions are in the same jvm, I am not sure it would always be the case (for instance in testng by default we fork the jvm)